### PR TITLE
[Separation] Separation of devkit and azure

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Move devkits and azures

image
image
image

We have some critical issues when install streamlit with poetry. And pf scripts, will fix in later patches.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -25,18 +25,21 @@
     "**/*.txt",
     ".gitignore",
     "scripts/docs/_build/**",
-    "src/promptflow/promptflow/azure/_restclient/flow/**",
-    "src/promptflow/promptflow/azure/_restclient/swagger.json",
-    "src/promptflow/promptflow/azure/_models/**",
+    "src/promptflow-azure/promptflow/azure/_restclient/flow/**",
```
Comment (wangchao1230) [inline .cspell.json:28]: shall this be changed when in the other azure PR?

Comment (crazygao) [reply .cspell.json:28]: let me make it in this PR

File: src/promptflow-devkit/promptflow/contracts/flow.py [hunk 2]
```diff
@@ -546,21 +546,13 @@ def get_environment_variables_with_overrides(
...
 
-    def get_connection_names(self, environment_variables_overrides: Dict[str, str] = None):
-        """Return connection names with environment variables overrides.
-        Note: only environment variables exist in flow.environment_variables will be considered.
-
-        :param environment_variables_overrides: used to override flow's environment variables.
-        :return: connection names used in this flow.
-        """
-        environment_variables_overrides = environment_variables_overrides or {}
+    def get_connection_names(self):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/contracts/flow.py:549]: why we changed this file?

File: src/promptflow-devkit/promptflow/contracts/tool.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from enum import Enum
 from typing import Any, Dict, List, Optional, Type, TypeVar
 
-from promptflow._constants import CONNECTION_NAME_PROPERTY, CONNECTION_DATA_CLASS_KEY
+from promptflow._constants import CONNECTION_NAME_PROPERTY
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/contracts/tool.py:11]: why change here?

File: src/promptflow-devkit/promptflow/contracts/flow.py (hunk 2 above)
Comment (crazygao) [reply src/promptflow-devkit/promptflow/contracts/flow.py:549]: Merge error. let me fix it

File: src/promptflow-devkit/promptflow/contracts/tool.py (hunk 3 above)
Comment (crazygao) [reply src/promptflow-devkit/promptflow/contracts/tool.py:11]: All merge erros.

File: src/promptflow-devkit/pyproject.toml [hunk 4]
```diff
@@ -36,6 +42,63 @@ packages = [
...
+opentelemetry-exporter-otlp-proto-http = ">=1.22.0,<2.0.0"  # trace support
+flask-restx = ">=1.2.0,<2.0.0"  # PFS Swagger
+flask-cors = ">=4.0.0,<5.0.0"  # handle PFS CORS
+promptflow-tracing = ">=1.0.0"  # tracing capabilities
+pyinstaller = ">=5.13.2"
+streamlit = ">=1.26.0"
+streamlit-quill = "<0.1.0"
+bs4 = "*"
+azure-identity = ">=1.12.0,<2.0.0"
+azure-ai-ml = ">=1.14.0,<2.0.0"
```
Comment (wangchao1230) [inline src/promptflow-devkit/pyproject.toml:73]: These should be azure dependency?

File: src/promptflow-recording/pyproject.toml [hunk 5]
```diff
@@ -37,6 +37,7 @@ packages = [
 [tool.poetry.dependencies]
 python = "<4.0,>=3.8"
 vcrpy = ">=5.1"
+promptflow-tracing = ">=0.1.0b1"
```
Comment (wangchao1230) [inline src/promptflow-recording/pyproject.toml:40]: >=0.1.0b1, <2.0.0?

File: src/promptflow/MANIFEST.in [hunk 6]
```diff
@@ -1,5 +1 @@
 include promptflow/azure/resources/*
-include promptflow/_sdk/_service/static/*
```
Comment (wangchao1230) [inline src/promptflow/MANIFEST.in:2]: remove the azure one?

File: src/promptflow-azure/pyproject.toml [hunk 7]
```diff
@@ -36,6 +36,12 @@ packages = [
 
 [tool.poetry.dependencies]
 python = "<4.0,>=3.8"
+azure-core = ">=1.26.4,<2.0.0"
```
Comment (wangchao1230) [inline src/promptflow-azure/pyproject.toml:39]: This should depends on promptflow-devkit?

File: src/promptflow-recording/pyproject.toml (hunk 5 above)
Comment (crazygao) [reply src/promptflow-recording/pyproject.toml:40]: OK

File: src/promptflow/MANIFEST.in (hunk 6 above)
Comment (crazygao) [reply src/promptflow/MANIFEST.in:2]: Yep

File: src/promptflow-devkit/pyproject.toml (hunk 4 above)
Comment (crazygao) [reply src/promptflow-devkit/pyproject.toml:73]: leave to promptflow repo.

File: src/promptflow-azure/pyproject.toml (hunk 7 above)
Comment (crazygao) [reply src/promptflow-azure/pyproject.toml:39]: OK

Comment (crazygao) [reply src/promptflow-azure/pyproject.toml:42]: yes.

File: src/promptflow/setup.py [hunk 8]
```diff
@@ -91,7 +65,7 @@
             "azureml-ai-monitoring>=0.1.0b3,<1.0.0",
         ],
         "executor-service": [
-            "fastapi>=0.109.0,<1.0.0",  # used to build web executor server
+            "promptflow-core[executor-service]",  # used to build web executor server
         ],
     },
     packages=find_packages(),
```
Comment (brynn-code) [inline src/promptflow/setup.py:71]: Move entry points.

Comment (crazygao) [reply src/promptflow/setup.py:71]: OK

File: src/promptflow-azure/pyproject.toml [hunk 9]
```diff
@@ -36,6 +39,13 @@ packages = [
...
+"azure-storage-blob[aio]" = ">=12.17.0,<13.0.0"  # add [aio] for async run download feature
+azure-identity = ">=1.12.0,<2.0.0"
+azure-ai-ml = ">=1.14.0,<2.0.0"
+azure-cosmos = ">=4.5.1,<5.0.0"  # used to upload trace to cloud
+pyjwt = ">=2.4.0,<3.0.0"  # requirement of control plane SDK
+promptflow-devkit = "*"
```
Comment (wangchao1230) [inline src/promptflow-azure/pyproject.toml:48]: >xx < 2?

File: src/promptflow-devkit/pyproject.toml [hunk 10]
```diff
@@ -36,6 +42,51 @@ packages = [
 
 [tool.poetry.dependencies]
 python = "<4.0,>=3.8"
+promptflow-core = "*"
```
Comment (wangchao1230) [inline src/promptflow-devkit/pyproject.toml:45]: specify range here

File: src/promptflow-devkit/promptflow/contracts/flow.py [hunk 11]
```diff
@@ -546,21 +546,13 @@ def get_environment_variables_with_overrides(
                 environment_variables[k] = v
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/contracts/flow.py:546]: contracts should be in core

Comment (crazygao) [reply src/promptflow-devkit/promptflow/contracts/flow.py:546]: good catch

File: src/promptflow-azure/pyproject.toml (hunk 9 above)
Comment (crazygao) [reply src/promptflow-azure/pyproject.toml:48]: OK

**Review by brynn-code — DISMISSED** [review]

File: src/promptflow/setup.py [hunk 12]
```diff
@@ -69,13 +69,7 @@
...
-    scripts=["pf", "pf.bat"],
-    entry_points={
-        "console_scripts": [
-            "pfazure = promptflow.azure._cli.entry:main",
-            "pfs = promptflow._sdk._service.entry:main",
-        ],
-    },
+    scripts=["pf.bat"],
```
Comment (wangchao1230) [inline src/promptflow/setup.py:72]: may need to take a look at this in next PR

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/setup.py (hunk 12 above)
Comment (crazygao) [reply src/promptflow/setup.py:72]: yes, I had a discussion with ying and she may take some actions on packaging and upgrading issue along with me.

**Review by brynn-code — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**